extension/framework/runner/environment.go

```diff
@@ -233,7 +233,7 @@ func (env *Environment) addTestingAPI() {
 func (env *Environment) getTransaction(isNew bool, options *transaction.TxOptions) transaction.Transaction {
 	if !isNew {
 		for _, tx := range env.dbTransactions {
-			if !tx.Closed() {
+			if !tx.Closed() && tx.GetIsolationLevel() == options.IsolationLevel {
```

Comment:
this may result in creation of nested transaction when isolation levels don't match. I think we should throw error instead creating "nested" one. Not truly nested second transaction can freeze while waiting for locks from the first one.